lib/node.js

```diff
@@ -7,24 +7,36 @@ const KEY = Buffer.alloc(16)
 
 module.exports = Node
 
+const Flags = {
+  HIDDEN: 1
+}
+
 function Node (data, seq, enc) {
   this.seq = seq || 0
   this.key = normalizeKey(data.key)
   this.value = data.value !== undefined ? data.value : null
   this.keySplit = split(this.key)
-  this.hidden = data.hidden || (data.flags ? !!(data.flags & 1) : false)
+  this.flags = data.flags || 0
   this.hash = hash(this.keySplit)
   this.trie = data.trieBuffer ? trie.decode(data.trieBuffer) : (data.trie || [])
   this.trieBuffer = null
   this.valueBuffer = data.valueBuffer || null
   this.length = this.hash.length * 4 + 1 + 1
   this.valueEncoding = enc
 }
+Node.Flags = Flags
 
 Node.prototype[inspect] = function (depth, opts) {
   return ((opts && opts.stylize) || defaultStylize)({seq: this.seq, key: this.key, value: this.value}, 'object')
 }
 
+Object.defineProperty(Node.prototype, 'hidden', {
+  enumerable: true,
+  get: function () {
+    return this.flags && this.flags & Flags.HIDDEN
```

Comment:
you don't need to guard against .flags. also return !!(this.flags & Flags.HIDDEN)